src/main/java/com/microsoft/graph/Requests/GraphClientFactory.java

```diff
@@ -0,0 +1,74 @@
+package com.microsoft.graph.Requests;
+
+import com.microsoft.graph.httpcore.CompressionHandler;
+import com.microsoft.graph.httpcore.GraphTelemetryHandler;
+import com.microsoft.kiota.RequestInformation;
+import com.microsoft.kiota.http.KiotaClientFactory;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+
+import javax.annotation.Nonnull;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.Format;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+/**
+ * The GraphClientFactory used to create the OkHttpClient.
+ */
+public class GraphClientFactory {
+    private GraphClientFactory() { }
+
+    /**
+     * The default OkHttpClient Builder for Graph.
+     *
+     * @return an OkHttpClient Builder instance.
+     */
+    public static OkHttpClient.Builder create() {
+        return create(new GraphClientOptions());
+    }
+
+    /**
+     * OkHttpClient Builder for Graph with specified Interceptors and optional GraphClientOptions.
+     *
+     * @param interceptors desired interceptors for use in requests.
+     * @param graphClientOptions the GraphClientOptions for use in requests.
+     * @return an OkHttpClient Builder instance.
+     */
+    public static OkHttpClient.Builder create(@Nonnull Interceptor[] interceptors, @Nullable GraphClientOptions graphClientOptions) {
```

Comment:
Maybe provide an overload with only interceptors? and switch the parameters here to nonnull